scripts/apps/search/directives/SearchParameters.ts

```diff
@@ -80,10 +80,17 @@ export function SearchParameters($location, asset, tags, metadata, common, desks
                 scope.params = params.params ? JSON.parse(params.params) : {};
 
                 if (scope.query && !Object.keys(scope.meta).length) {
-                    let queryTerms = scope.query.split(/(?!\(.*)\s(?![^(]*?\))/g);
+                    // split scope.query by spaces outside the parentheses
+                    // ex: "slugline:(abc) headline:(xyz)" will become ["slugline:(abc)", "headline:(xyz)"]
+                    const queryTerms = scope.query.split(/(?!\(.*)\s(?![^(]*?\))/g);
```

Comment:
Use the internal data structures instead of trying to parse the same information from the string which is displayed to the user. The format which the information is displayed to the user should be allowed to change without breaking features.

Reply:
Sorry, I didn't get you.

Reply:
Strings like `slugline:(abc) headline:(xyz)` are created from objects with separate fields. You must not parse the string. If you need to read the data, read it from the original objects the string is made from, not from the string itself.

Here's an example of where a similar string is created
> `tags.selectedParameters.push(tag(value + ':(' + name + ')'))`

Reply:
It will work only for those fields which are kind of `cv` like `subject` or `company codes`. Whereas `slugline`, `headline` or `body` are hardcoded strings.